### ISSUE TITLE
Getting eigenfunction gradients

### ISSUE THREAD

**Issue by nec4** [hsidky/srv#2]
Hello. I am enjoying the tools, but I am not familiar with keras/tensorflow. Is there a way to get get the gradients of the SRV eigenfunctions with respect to the SRV network input - that is, to get d(network_output)/d(input_features)? Thanks.

**Comment by hsidky** [comment]
Hi! 
Sorry for the late reply! This is how you can do it with keras/tensorflow. The code below is done for each eigenfunction spearately, and I've defined the index of the eigenfunction  using the variable `component`. You can adjust accordingly and do this for as few or as many components as you'd like.

```
import tensorflow as tf
import keras.backend as K

component = 0

grads = tf.gradients(tf.multiply(-1.0,srv.encoder.output[:,component]), srv.encoder.input)
grad_model = K.function(srv.encoder.input, grads)

# Evaluate gradient
gradient = grad_model(x_values)[0]
```

If you run into any issues let me know and I'll be happy to  share an explicit example. 

Hythem

**Comment by nec4** [comment]
Thanks! I will try this out.

**Comment by nec4** [comment]
Seems to work well for a 1D test - Thank you for the response.

**Comment by hsidky** [comment]
Great. I'm closing this issue. If you still have any questions feel free to reopen this or create a new one.
